core/server/models/client.js

```diff
@@ -10,6 +10,7 @@ Client = ghostBookshelf.Model.extend({
     tableName: 'clients',
 
     defaults: function defaults() {
+        // @TODO: we cannot delete this ugly check here, because ALL routing tests rely on a static client secret
```

Comment:
This would take too long to fix. All routing tests expect that the clients have `not_available` as client secret.
There are workarounds possible, but they are ugly as well. For example:

`config.get('clientSectet') || crypto.randomBytes...`

All testing env's could set the `not_available` secret.
But this is still weird and not clean.
I can only see one solution and this is: make the routing tests better.